board/drivers/gpio.h

```diff
@@ -58,7 +58,7 @@ void set_gpio_alternate(GPIO_TypeDef *GPIO, unsigned int pin, unsigned int mode)
 void set_gpio_pullup(GPIO_TypeDef *GPIO, unsigned int pin, unsigned int mode) {
   ENTER_CRITICAL();
   uint32_t tmp = GPIO->PUPDR;
-  tmp &= ~(3U << (pin * 2U));
+  tmp &= ~((uint32_t)3U << ((uint32_t)(pin * 2U)));
```

Comment:
Sorry for this. MISRA decided to validate this path when we have mads_exit_controls() due to LAG. But also on the braking... Like, I dont know why but commenting either makes MISRA happy and doesn't complain here. Adding them somehow makes MISRA reconsider it's life choices and comes and checks something unrelated....

Reply:
Resolved and reverted in https://github.com/sunnypilot/panda/commit/6dab7bbb4787aa13a17242bb4c445a18c67d81c7.